lib/SPIRVProducerPass.cpp

```diff
@@ -5196,7 +5278,30 @@ void SPIRVProducerPass::HandleDeferredInstruction() {
 
         FunctionType *CalleeFTy = cast<FunctionType>(Call->getFunctionType());
         for (unsigned i = 0; i < CalleeFTy->getNumParams(); i++) {
-          Ops << MkId(VMap[Call->getOperand(i)]);
+          auto *operand = Call->getOperand(i);
+          if (operand->getType()->isPointerTy()) {
+            auto sc =
+                GetStorageClass(operand->getType()->getPointerAddressSpace());
+            if (sc == spv::StorageClassStorageBuffer) {
+              // Passing SSBO by reference requires variable pointers storage
+              // buffer.
+              setVariablePointersStorageBuffer(true);
+            } else if (sc == spv::StorageClassWorkgroup) {
+              // Workgroup references require variable pointers if they are not
+              // memory object declarations.
+              if (auto *operand_call = dyn_cast<CallInst>(operand)) {
+                // Workgroup accessor represents a variable reference.
+                if (!operand_call->getCalledFunction()->getName().startswith(
+                        clspv::WorkgroupAccessorFunction()))
+                  setVariablePointers(true);
+              } else {
+                // Arguments are function parameters.
+                if (!isa<Argument>(operand))
```

Comment:
I guess if the pointer is a workgroup variable, then it shows up as a call to the workgroup accessor function.
So I see how this implements the condition.  ok